packages/twenty-front/src/pages/object-record/RecordIndexPage.tsx

```diff
@@ -81,13 +85,19 @@ export const RecordIndexPage = () => {
             <StyledIndexContainer>
               <ContextStoreComponentInstanceContext.Provider
                 value={{
-                  instanceId: 'record-index',
+                  instanceId: `record-index-${objectMetadataItem.id}`,
                 }}
               >
-                <RecordIndexContainerContextStoreObjectMetadataEffect />
-                <RecordIndexContainerContextStoreNumberOfSelectedRecordsEffect />
-                <SetMainContextStoreComponentInstanceIdEffect />
-                <RecordIndexContainer />
+                <ActionMenuComponentInstanceContext.Provider
+                  value={{
+                    instanceId: `record-index-${objectMetadataItem.id}`,
+                  }}
+                >
+                  <RecordIndexContainerContextStoreObjectMetadataEffect />
+                  <RecordIndexContainerContextStoreNumberOfSelectedRecordsEffect />
+                  <MainContextStoreComponentInstanceIdSetterEffect />
```

Comment:
missing setters in RightDrawer + ShowPage (objectMetadataSetter + main context store)